### ISSUE TITLE
Out-Default topic lacks proper framing

### ISSUE THREAD

**Issue by mklement0** [MicrosoftDocs/PowerShell-Docs#5131]
I think the following information (excerpted from [this Stack Overflow](https://stackoverflow.com/a/58958346/45375) answer) could help readers make better sense of the purpose of `Out-Default` and when it is needed / appropriate to use:

----------

**[`Out-Default`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/out-default) is _not_ meant to be called from user code.**

Instead, it is meant to be used only by [PowerShell host applications](https://docs.microsoft.com/en-us/powershell/scripting/developer/hosting/writing-a-windows-powershell-host-application), including PowerShell itself.
It creates pretty-printed string representations of the output objects using [PowerShell's output-formatting system](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_format.ps1xml) and sends them to the host for display.

The reason that it is a public cmdlet is that you are allowed to *override* it in order to define a *custom* output formatter, although doing so is rare.

Overriding is as simple as defining an advanced `Out-Default` function in your session, which must have the same parameter declarations and pipeline-binding behavior. You can scaffold such a function via a *wrapper (proxy) function*, as  demonstrated in the bottom section of [this answer](https://stackoverflow.com/a/55539863/45375)). Note that in order to produce visible output from your function, you must send your custom-formatted strings either to the original `Out-Default` or use the _host APIs_ to print them (`Write-Host`, in the simplest case). 

-----------

Note that an older version of this help topic actually had a more helpful (albeit too terse) summary than the current version:

> This cmdlet has no effect on the formatting or output of PowerShell commands. It
    is a placeholder that lets you write your own Out-Default function or cmdlet.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ee6e614d-658d-32ac-dc39-e05e49e3da91
* Version Independent ID: aa4edc10-7fe1-7252-461a-998b6223f02e
* Content: [Out-Default (Microsoft.PowerShell.Core)](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/out-default?view=powershell-7)
* Content Source: [reference/7/Microsoft.PowerShell.Core/Out-Default.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7/Microsoft.PowerShell.Core/Out-Default.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

**Comment by sdwheeler** [comment]
@mklement0 I am working on an update. But I am confused by one of your statements:

> for now to suppress display output for all streams, you must use redirection *>&1, 
> but note that this only works with external programs, not with PowerShell commands.

You say this only work with external programs but then you show a working PowerShell example. Am I missing something?

Also, in a related note, see https://github.com/MicrosoftDocs/PowerShell-Docs/wiki/Output-missing-from-PowerShell-transcript

This is another case for explicity use of `Out-Default`.

**Comment by mklement0** [comment]
Thanks, @sdwheeler. 

I suggest holding off on an update until we have clarity on whether `-Transcript` ever makes sense for end users.

As for https://github.com/MicrosoftDocs/PowerShell-Docs/wiki/Output-missing-from-PowerShell-transcript

The root cause here is the asynchronous behavior of the _implicitly_ applied `Format-Table` formatting, as described in [this Stack Overflow answer](https://stackoverflow.com/a/43691123/45375).

However, I would _not_ recommend `Out-Default` to solve the sequencing problem.

Instead, we should recommend `Out-Host`, which does have legitimate uses.

**Comment by mklement0** [comment]
@sdwheeler, please see the update to the OP, arrived at after the discussion in https://github.com/PowerShell/PowerShell/issues/11135 (I've invited participants there to review).

**Comment by mklement0** [comment]
@sdwheeler, regarding using `Out-Host` instead of `Out-Default` to solve the output-sequencing (-timing) discussed above:

`Out-Host` is still the better choice, but, like `Out-Default`, it has the side effect of suppressing success output (by redirecting it to the host).

This means that if you use `Out-Host` to solve your output-sequencing problem - which is not limited to _transcripts_, by the way - a caller of your script won't be able to capture the command's output.

In short: Use the `Out-Host` workaround only if you are willing to assume that your command's output needs no further programmatic processing and should go to the display (host) only.
